src/runtime/button-api-test.js

```diff
@@ -183,6 +186,63 @@ describes.realWin('ButtonApi', {}, env => {
     expect(button.getAttribute('title')).to.equal('S\'abonner avec Google');
   });
 
+  it('should log button impression on create.', () => {
+    eventManagerMock.expects('logEvent').withExactArgs(
+        CreateButtonRelatedClientEvent(
+            AnalyticsEvent.IMPRESSION_SUBSCRIBE_BUTTON)).once();
+    buttonApi.create(handler);
+  });
+
+  it('should log button impression on attach.', () => {
+    eventManagerMock.expects('logEvent').withExactArgs(
+        CreateButtonRelatedClientEvent(
+            AnalyticsEvent.IMPRESSION_SUBSCRIBE_BUTTON)).once();
+    const button = doc.createElement('button');
+    buttonApi.attach(button, {}, handler);
+  });
+
+  it('should log paywall impression on create.', () => {
+    eventManagerMock.expects('logEvent').withExactArgs(
+        CreateButtonRelatedClientEvent(
+            AnalyticsEvent.IMPRESSION_SUBSCRIBE_BUTTON)).once();
+    eventManagerMock.expects('logEvent').withExactArgs(
+        CreateButtonRelatedClientEvent(
+            AnalyticsEvent.IMPRESSION_PAYWALL)).once();
+    buttonApi.create(handler, null, true);
+  });
+
+  it('should log paywall impression on attach.', () => {
+    eventManagerMock.expects('logEvent').withExactArgs(
+        CreateButtonRelatedClientEvent(
+            AnalyticsEvent.IMPRESSION_SUBSCRIBE_BUTTON)).once();
+    eventManagerMock.expects('logEvent').withExactArgs(
+        CreateButtonRelatedClientEvent(
+            AnalyticsEvent.IMPRESSION_PAYWALL)).once();
+    const button = doc.createElement('button');
+    buttonApi.attach(button, {}, handler, true);
+  });
+
+  it('should log offers impression on create.', () => {
```

Comment:
I'm just wondering why it logs offers impression when you create it.  Don't they have to click it to view offers?  Or if it has a default offer there is no offers impression?